.github/workflows/build.yml

```diff
@@ -206,7 +206,6 @@ jobs:
           path: coverage.xml
 
   sonarcloud:
-    needs: [test]
```

Comment:
We will want to wait for `test` to execute before running this Job. Perhaps doing this might work?

```
    needs: [test]
    if: always()
```

Reply:
As the description "Use jobs.<job_id>.needs to identify any jobs that must complete successfully before this job will run." I think needs: [test] will require the test run through before starting.

Reply:
Yes - However if you include `always` this will be run even if `test` fails, however, it should wait until `test` fails to run this `sonarcloud` job.

Reply:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds

> If you would like a job to run even if a job it is dependent on did not succeed, use the always() conditional expression in jobs.<job_id>.if.